hazelcast/src/main/java/com/hazelcast/config/PermissionConfig.java

```diff
@@ -27,8 +29,8 @@
     private PermissionType type;
     private String name;
     private String principal;
-    private Set<String> endpoints = new HashSet<String>();
-    private Set<String> actions = new HashSet<String>();
+    private Set<String> endpoints = newSetFromMap(new ConcurrentHashMap<String, Boolean>());
```

Comment:
what was the need for concurrent access? I expect no concurrent access to config classes at first glance.

Reply:
There was a wrong usage in the `DefaultPermissionPolicy` (see hazelcast/hazelcast-enterprise#2804)